File: app/frontend/src/types/module.ts
/** Module and lesson types */
export type Track = 'user' | 'analyst' | 'developer' | 'architect';

export interface Module {
  id: number;
  title: string;
  description: string | null;
  track: Track;
  order_index: number;
  duration_hours: number;
  prerequisites: number[] | null;
  learning_objectives: string[] | null;
  is_published: boolean;
}

export interface Lesson {
  id: number;
  module_id: number;
  title: string;
  content: string;
  order_index: number;
  estimated_minutes: number | null;
  lesson_type: string;
}

export interface ModuleDetail extends Module {
  lessons: Lesson[];
  has_assessment: boolean;
}

File: app/backend/schemas/module.py
"""Module and lesson schemas"""
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from app.backend.models.module import Track


class LessonResponse(BaseModel):
    """Lesson response schema"""
    id: int
    module_id: int
    title: str
    content: str
    order_index: int
    estimated_minutes: Optional[int]
    lesson_type: str
    
    class Config:
        from_attributes = True


class ModuleResponse(BaseModel):
    """Module response schema"""
    id: int
    title: str
    description: Optional[str]
    track: Track
    order_index: int
    duration_hours: float
    prerequisites: Optional[List[int]]
    learning_objectives: Optional[List[str]]
    is_published: bool
    
    class Config:
        from_attributes = True


class ModuleDetailResponse(BaseModel):
    """Module detail with lessons"""
    id: int
    title: str
    description: Optional[str]
    track: Track
    order_index: int
    duration_hours: float
    prerequisites: Optional[List[int]]
    learning_objectives: Optional[List[str]]
    lessons: List[LessonResponse]
    has_assessment: bool = False  # Whether module has assessments
    
    class Config:
        from_attributes = True


class ModuleListResponse(BaseModel):
    """List of modules"""
    modules: List[ModuleResponse]
    total: int


